Program.cs

```diff
@@ -146,6 +146,29 @@
 //post follower
 //delete follower
 
+
+
+app.MapGet(
+    "/pirates/{id}",
+    (int? id) =>
+    {
+        Pirate pirate = pirates.FirstOrDefault(p => p.Id == id);
+
+        return Results.Ok(new PirateDTO
```

Comment:
If FirstOrDefault doesn't return a pirate, this will give an error as pirate would not have properties being null. Return a bad request result if pirate is null.